src/brod_topic_subscriber.erl

```diff
@@ -299,6 +330,23 @@ subscribe_partition(Client, Topic, Consumer) ->
       end
   end.
 
+handle_message_set(Partition, MessageSet, State) ->
+  #state{cb_fun = CbFun, cb_state = CbState} = State,
+  ErlRef = erlang:make_ref(),
+  AckRef = {Partition, ErlRef},
```

Comment:
`AckRef` should be `{partition(), offset()}`
Here it should be the offset of the last message in the received message set
(you may safely call `lists:last/1` because message-set is never empty)

Also, there is no need to evaluate `lists:last/1` when `AckNow` is not true

Reply:
👍